src/components/notes/saving-indicator.jsx

```diff
@@ -19,6 +20,61 @@ const outOfSyncAfter = 30 * sec
  * @property {Ref} bannerRef - Where to display banners
  */
 
+/**
+ * Checks if the server is out of sync
+ * (our changes were not successfully sent and accepted for too much time)
+ * @param {object} params
+ * @param {boolean} params.isDirty - if there are changes to be sent
+ * @param {integer} params.dirtyAge - number of milliseconds waiting
+ *                                    with changes to be sent
+ * @returns {boolean}
+ */
+function checksOutOfSync({ isDirty, dirtyAge }) {
+  return isDirty && dirtyAge && dirtyAge > outOfSyncAfter
+}
+
+/**
+ * Checks for disconnection (browser offline or out of sync)
+ *
+ * @param {object} params
+ * @param {boolean} params.isOutOfSync
+ * @returns {{isDisconnected, disconnectedAge}}
+ */
+function checksDisconnection({ isOutOfSync }) {
```

Comment:
Since it is a hook, I think it would be better to keep the `use` convention. useDisconnection ?

Reply:
We could, but I found the name not really clear, and as the function is in the same file, I did think it wasn't as important.